### PR TITLE
fix: circular dependency between Permit, Applications, Payments

### PR DESCRIPTION
**Pull request by praju-aot**
# Description

Please provide a summary of the change and the issue fixed. Please include relevant context. List dependency changes.

Fixes # (issue)
fixes Circular dependency between Permit, Applications, Payments
## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] npm run test
- [x] npm run test:e2e
- [x] Manual Unit Testing - Create Application by CVClient and Staff, Issue By CV Client and Staff, Amend by Staff, Void/Revoke by Staff, Delete by CV Client.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1268-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1268-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1268-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1268-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

File: vehicles/src/common/enum/application-status.enum.ts [hunk 1]
```diff
@@ -16,22 +16,23 @@ export enum ApplicationStatus {
...
-    ApplicationStatus.WAITING_PAYMENT,
-    ApplicationStatus.CANCELLED,
-    ApplicationStatus.UNDER_REVIEW,
-  ];
+export const ACTIVE_APPLICATION_STATUS_FOR_ISSUANCE: ReadonlyArray<ApplicationStatus> =
+  [ApplicationStatus.IN_PROGRESS, ApplicationStatus.WAITING_PAYMENT];
 
 /**
- * Apllication statuses to be considered for Application In Progress (AIP) from CV Client POV.
+ * Apllication statuses to be considered for Application In Progress (AIP) Tab for Cv Client Users.
```
Comment (krishnan-aot) [inline vehicles/src/common/enum/application-status.enum.ts]: Application* everywhere

File: vehicles/src/app.module.ts [hunk 2]
```diff
@@ -91,8 +92,9 @@ const envPath = path.resolve(process.cwd() + '/../');
     CommonModule,
     PendingUsersModule,
     AuthModule,
-    PermitModule,
     PaymentModule,
+    ApplicationModule, //! Application Module should be imported before PermitModule to avoid URI conflict
```
Comment (krishnan-aot) [inline vehicles/src/app.module.ts:96]: if shopping cart has its own module, will its URIs clash with ApplicationModule and in turn rely on finding the right spot for placing it in this list?
This is the cart URI: `/permit/applications/shopping-cart`. Maintaining a delicate balance is somewhat of an overhead.

File: vehicles/src/common/helper/database.helper.ts [hunk 3]
```diff
@@ -5,10 +5,16 @@ export const callDatabaseSequence = async (
...
-    'Select NEXT VALUE FOR ' + databaseSequenceName + ' as Next_Value;';
-  const sequence = (await queryRunner.query(query)) as [{ Next_Value: string }];
-  return sequence[0].Next_Value;
+  let sequence: string;
+  try {
+    const query =
+      'Select NEXT VALUE FOR ' + databaseSequenceName + ' as Next_Value;';
+    const result = (await queryRunner.query(query)) as [{ Next_Value: string }];
+    sequence = result?.at(0).Next_Value;
+  } finally {
```
Comment (krishnan-aot) [inline vehicles/src/common/helper/database.helper.ts:14]: Does it need a catch block?

File: vehicles/src/app.module.ts (hunk 2 above)
Comment (praju-aot) [reply vehicles/src/app.module.ts:96]: Cart URI will have to be changed accordingly as its yet not implemented.

File: vehicles/src/common/helper/database.helper.ts (hunk 3 above)
Comment (praju-aot) [reply vehicles/src/common/helper/database.helper.ts:14]: No, it does not need a catch block. Error on adding catch block

**Review by krishnan-aot — APPROVED** [review]

**Pull request merged**